src/librustc_middle/mir/mono.rs

```diff
@@ -229,6 +204,19 @@ impl<'a, 'tcx> HashStable<StableHashingContext<'a>> for MonoItem<'tcx> {
     }
 }
 
+impl<'tcx> fmt::Display for MonoItem<'tcx> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        ty::tls::with(|tcx| match tcx.lift(self).expect("could not lift for printing") {
+            MonoItem::Fn(instance) => {
+                let name = tcx.def_path_str_with_substs(instance.def_id(), instance.substs);
+                write!(f, "fn {}", &name)
+            }
+            MonoItem::Static(def_id) => write!(f, "static {}", &tcx.def_path_str(def_id)),
```

Comment:
This could just use `Instance::new(def_id, Substs::empty())` (assuming the other comment applies)

Reply:
Fixed!